call.go

```diff
@@ -110,3 +113,163 @@ func WithRoutingKey(rk string) CallOption {
 func WithRoutingDelegate(rd string) CallOption {
 	return CallOption{func(o *OutboundCall) { o.routingDelegate = &rd }}
 }
+
+// InboundCall is an incoming call. It holds information about the inbound
+// call and its response.
+//
+// Encoding authors may use InboundCall to provide information about the
+// incoming request on the Context and receive response headers through
+// WriteResponseHeader.
+type InboundCall struct {
+	ctx        context.Context
+	resHeaders []keyValuePair
+	req        *transport.Request
+}
+
+type inboundCallKey struct{} // context key for *InboundCall
+
+// NewInboundCall builds a new InboundCall with the given context.
+//
+// A new context is returned and must be used in place of the original.
+func NewInboundCall(ctx context.Context) (context.Context, *InboundCall) {
+	call := &InboundCall{ctx: ctx}
+	return context.WithValue(ctx, inboundCallKey{}, call), call
+}
+
+// getInboundCall returns the inbound call on this context or nil.
+func getInboundCall(ctx context.Context) *InboundCall {
+	if call, ok := ctx.Value(inboundCallKey{}).(*InboundCall); ok {
+		return call
+	}
+	return nil
```

Comment:
instead of returningjust a `*InboundCall`, if you return a `bool` as well, it'll make it more obvious at call sites that you need to check if you actually got a call or not.